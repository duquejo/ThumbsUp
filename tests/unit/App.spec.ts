import { describe, it, expect, beforeEach, vi } from 'vitest';
import { VueWrapper, shallowMount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import AppVue from '@/App.vue';

describe('+ AppVue unit test', () => {

  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = shallowMount(AppVue, {
      global: {
        plugins: [createTestingPinia({
          createSpy: vi.fn(),
        })],
      },
    });
  });
  
  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
