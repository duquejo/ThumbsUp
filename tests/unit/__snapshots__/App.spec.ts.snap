// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`+ AppVue unit test > should render properly 1`] = `
"<transition-stub data-v-7a7a37b1="" name="slide-fade" appear="false" persisted="true" css="true">
  <div data-v-7a7a37b1="" style="display: none;">
    <navigation-component-stub data-v-7a7a37b1=""></navigation-component-stub>
    <hero-component-stub data-v-7a7a37b1=""></hero-component-stub>
    <general-layout-stub data-v-7a7a37b1=""></general-layout-stub>
  </div>
</transition-stub>"
`;
